main.py

```diff
@@ -244,14 +244,17 @@ def strip_commits(commits, type_regex):
         dict: selected commits of every scope.
     '''
     # TODO: add an attribute to ignore scope 
-    regex = r'^'+ type_regex + r'[(](.+?)[)]'
+    regex = r'^' + type_regex + r'(?:[(](.+?)[)])?'
     scopes = {}
     for commit in commits:
-        if re.match(regex, commit['head']):
-            scope = re.findall(regex, commit['head'])[0]
-            if scope.lower() == 'changelog' and regex == r'^docs[(](.+?)[)]':
+        head = commit['head']
+        if re.match(regex, head):
+            scope = re.findall(regex, head)[0]
+            if scope == '':
+                scope = 'all'
```

Comment:
@BobAnkh look closely at this line: I just throw empty scopes in a 'all' scope here.
There are two solutions

1. don't touch the scope here and handle it smart during rendering (so no subsection with all)
2. Leave it like it is and accept that we have a sub-section with `all`

Reply:
Maybe it's better to leave it empty and don't open a sub section for them

Reply:
Is it a good idea to put this kinds of commits in a sub-section with `*` instead of `all`?

> Maybe it's better to leave it empty and don't open a sub section for them

Can you explain it to me in more details? Do you mean this :

```markdown
-
  - add xxx
  - remove xxx
```

which renders that:

-
  - add xxx
  - remove xxx

I suppose it could not pass the markdown lint check?

Maybe a catch-all subsection `*`? (Because single * character cannot be used in list, so I use `*`)

- `*`
	- add xxx
	- remove xxx

Do you think it a good choice? If this is not a good idea, I suppose to just add them at the same level of scope.

Reply:
Yeah, Im also not quite sure. An alternative to `all` could be `general`. Even better if we add a action configuration for the general section which defaults to `general` or `all` so then the user can decide under which section these items should occur.

What you showing are lists of lists and what I mean is rendering them directly als list:

e.g. now

- all
  - some general commit
  - some general commit
- Bugfixes
  - some fix commit
  - some fix commit
 - Features
   - some feat commit

an this would be as simple list:

- some general commit
- some general commit
- Bugfixes
  - some fix commit
  - some fix commit
 - Features
   - some feat commit

But looking at these now it might look cleaner if we have this catch-em-all section:

- General
  - some general commit
  - some general commit
- Bugfixes
  - some fix commit
  - some fix commit
 - Features
   - some feat commit

Reply:
oh my bad the types a sections and the lists elements are the scopes...

Reply:
Ok, then can we set a action configuration with a default empty string, meaning not showing theses commits? And if this param is set to a non-empty string, will all these commits without scope be listed under this scope? In this way, we will not have breaking changes compared to previous versions and users can also show commits without scope if they want?

Reply:
here again the variants:

e.g. now

- all
  - some general commit
  - some general commit
### Bugfixes
- somescope
  - some fix commit
  - some fix commit
### Features
- all
  - some feat commit
- somescope
  
an this would be as simple list:

- some general commit
- some general commit
### Bugfixes
- somescope
  - some fix commit
- someotherscope
  - some fix commit
### Features
- somescope 
  - some feat commit

Reply:
... The second option would then do as your suggested and discard those commits...

Reply:
Great. I agree with you. So can you help me implement it?

Reply:
like that 
```
  DEFAULT_SCOPE:
    description: "The scope under which unscoped commits can be found e.g. 'fix: some general fix'"
    required: false
    default: 'general'
  SUPPRESS_UNSCOPED:
    description: "Suppress the generation of release notes for un-scoped commits e.g. 'fix: some general fix'"
    required: false
    default: 'false'
```

Reply:
sure will do

Reply:
Thank you for your help. You can just doing the job and ignore all the suggestions proposed by bots. I will help to fix.

Reply:
no worries, I think the bots are fine... bit verbose but they do their job 😂

Reply:
changes pushed btw.